analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -137,43 +142,62 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
         managedFiles: Map<PackageManager, List<File>>,
         curationProvider: PackageCurationProvider
     ): AnalyzerResult {
-        val analyzerResultBuilder = AnalyzerResultBuilder(curationProvider)
-
-        runBlocking(Dispatchers.IO) {
-            managedFiles.map { (manager, files) ->
-                async {
-                    val results = manager.resolveDependencies(files, labels)
-
-                    // By convention, project ids must be of the type of the respective package manager. An exception
-                    // for this is Pub with Flutter, which internally calls Gradle.
-                    results.projectResults.forEach { (_, result) ->
-                        val invalidProjects = result.filterNot {
-                            val projectType = it.project.id.type
-
-                            projectType == manager.managerName ||
-                                    (manager.managerName == "Pub" && projectType == "Gradle")
-                        }
-
-                        require(invalidProjects.isEmpty()) {
-                            val projectString = invalidProjects.joinToString { "'${it.project.id.toCoordinates()}'" }
-                            "Projects $projectString must be of type '${manager.managerName}'."
-                        }
-                    }
+        val state = AnalyzerState(curationProvider)
+
+        runBlocking {
+            managedFiles.entries.map { (manager, files) ->
+                PackageManagerRunner(
+                    manager = manager,
+                    definitionFiles = files,
+                    labels = labels,
+                    mustRunAfter = emptySet(),
+                    finishedPackageManagersState = state.finishedPackageManagersState,
+                    onResult = { result -> state.addResult(manager, result) }
+                )
+            }.forEach { launch { it.start() } }
+        }
+
+        return state.buildResult()
+    }
+}
+
+private class AnalyzerState(curationProvider: PackageCurationProvider) {
+    private val builder = AnalyzerResultBuilder(curationProvider)
+    private val scope = CoroutineScope(Dispatchers.Default)
```

Comment:
What is the reason for switching from `Dispatchers.IO` to `Dispatchers.Default`?

Reply:
I'm using `Dispatchers.Default` here because the state itself does not do IO.